scripts/tools/metricProgrammable.yml

```diff
@@ -0,0 +1,490 @@
+version: "1.8"
+class: $tMetricProgrammable
+name: GetParamInfoExp
+decl: static
+details:
+    - "Returns information about the parameters of the metric programmable handle."
+params:
+    - type: $t_metric_programmable_exp_handle_t
+      name: hMetricProgrammable
+      desc: "[in] handle of the metric programmable"
+    - type: "uint32_t*"
+      name: pParameterCount
+      desc: |
+            [in,out] count of the parameters to retrieve parameter info.
+            if value pParameterCount is greater than count of parameters available, then pParameterCount will be updated with count of parameters available.
+    - type: "$t_metric_programmable_param_info_exp_t*"
+      name: pParameterInfo
+      desc: |
+            [out][range(1, *pParameterCount)] array of parameter info.
+            if parameterCount is less than the number of parameters available, then driver shall only retrieve that number of parameter info.
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Get the information about the parameter value of the metric programmable."
+version: "1.8"
+class: $tMetricProgrammable
+name: GetParamValueInfoExp
+decl: static
+details:
+    - "Returns the value-information about the parameter at the specific ordinal of the metric programmable handle."
+params:
+    - type: $t_metric_programmable_exp_handle_t
+      name: hMetricProgrammable
+      desc: "[in] handle of the metric programmable"
+    - type: uint32_t
+      name: parameterOrdinal
+      desc: "[in] ordinal of the parameter in the metric programmable"
+    - type: "uint32_t*"
+      name: pValueInfoCount
+      desc: |
+            [in,out] count of parameter value information to retrieve.
+            if value at pValueInfoCount is greater than count of value info available, then pValueInfoCount will be updated with count of value info available.
+    - type: "$t_metric_programmable_param_value_info_exp_t*"
+      name: pValueInfo
+      desc: |
+            [out][range(1, *pValueInfoCount)] array of parameter value info.
+            if pValueInfoCount is less than the number of value info available, then driver shall only retrieve that number of value info.
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Create metric handles by applying parameter values on the metric programmable handle."
+version: "1.8"
+class: $tMetric
+name: CreateFromProgrammableExp
+decl: static
+details:
+    - "Multiple parameter values could be used to prepare a metric."
+    - "If parameterCount = 0, the default value of the metric programmable would be used for all parameters"
+    - "Parameter values are updated to pParameterValues in the same order as retrieved using $tMetricProgrammableGetParamInfoExp"
+    - "If multiple metrics are generated, then implementation defined C string will be post-fixed to metric name and description"
```

Comment:
- if multiple metrics are created in a single call to the API? 
- "then implementation defined C string will be post-fixed to metric name and description"  does this mean at the user input will not be verbatim respected? why would this only apply to the case of multiple metrics?

Reply:
>"then implementation defined C string will be post-fixed to metric name and description" does this mean at the user input will not be verbatim respected? why would this only apply to the case of multiple metrics?

Corrected it. 
Implementations can post-fix C strings to metric names based on the parameter values set.